### PR TITLE
Replace boto_retry mechanism with boto3 standard retry mechanism

### PR DESCRIPTION
**Pull request by gockle**
*Issue #, if available:*
#307 

*Description of changes:*
1. Remove the boto_retry mechanism and replace with [boto3 standard retry mode](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html)
2. Remove models loaded into the lambda and use the default lambda service.json

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/lambda/util/boto_utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#                                                                                                                    #
+#  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES #
+#  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
+#  and limitations under the License.                                                                                #
+######################################################################################################################
+
+import boto
+import botocore
+
+def get_config():
```
Comment (hearde) [inline source/lambda/util/boto_utils.py:17]: Why is this here and in init?

File: source/lambda/util/boto_utils.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    user_agent_extra_string = os.getenv("USER_AGENT_EXTRA", "AwsSolution/SO0030/v1.5.0")
+    solution_config = {"user_agent_extra": user_agent_extra_string, "retries": {'max_attempts': 5, 'mode': 'standard'}}
+    return botocore.config.Config(**solution_config) 
+
+# def get_client(service, context=None, session=None):
```
Comment (hearde) [inline source/lambda/util/boto_utils.py:22]: Remove

File: source/lambda/util/boto_utils.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+######################################################################################################################
```
Comment (hearde) [inline source/lambda/util/boto_utils.py:1]: Use the short header

File: source/lib/aws-instance-scheduler-stack.ts [hunk 4]
```diff
@@ -860,7 +860,7 @@ export class AwsInstanceSchedulerStack extends cdk.Stack {
       ],
       effect: Effect.ALLOW,
       resources: [
-        cdk.Fn.sub("arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/Solutions/instance-scheduler/UUID/*")
+        cdk.Fn.sub("arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/Solutions/aws-instance-scheduler/UUID/*")
```
Comment (hearde) [inline source/lib/aws-instance-scheduler-stack.ts:863]: This seems unrelated, what's the purpose?

File: source/lambda/boto_retry/__init__.py [hunk 5]
```diff
@@ -11,136 +11,20 @@
...
-                            method_suffix=DEFAULT_SUFFIX):
+def get_client(service_name, region=None, session=None):
     """
     Creates a bot3 client for the specified service name and region. The return client will have additional method for the
     specified methods that are wrapped with the logic of the specified wait strategy or the default strategy for that service.
     The method names must be valid for the boto3 service client. The name of the added functions is the name of the original
     function plus the (default) value of method_suffix parameter
     :param service_name: Name of the service
-    :param methods: List of methods for which a new method will be added to the client wrapped in retry logic
     :param context: Lambda execution context
```
Comment (hearde) [inline source/lambda/boto_retry/__init__.py:25]: Looks like this param is gone too

File: source/lambda/boto_retry/__init__.py [hunk 6]
```diff
@@ -11,136 +11,20 @@
...
-    else:
-        retry_class = AwsApiServiceRetry
-    return retry_class
-
-
-def get_client_with_retries(service_name, methods, context=None, region=None, session=None, wait_strategy=None,
-                            method_suffix=DEFAULT_SUFFIX):
+def get_client(service_name, region=None, session=None):
     """
     Creates a bot3 client for the specified service name and region. The return client will have additional method for the
```
Comment (hearde) [inline source/lambda/boto_retry/__init__.py:20]: Should be updated - client with standard retry policy

**Review by hearde — APPROVED** [review]
When I commented "Should be updated - client with standard retry policy" I was just referring to the docstring for the `get_client` function. I think it's fine to leave it as `get_client`.

It looks like when you select a single line for comment in a gh pr, that line is the last line that appears above the comment in the summary view, and gh seems to choose how much context to give it, so looking at the summary it looks like I was commenting on the function name. I'll try to be more clear in the future, sorry about that.

File: source/lambda/configuration/config_admin.py [hunk 7]
```diff
@@ -108,8 +108,7 @@ def __init__(self, logger, context):
         :param context: Lambda context
         """
         self._table_name = self.table_name
-        self._table = boto3.resource("dynamodb").Table(self._table_name)
-        add_retry_methods_to_resource(self._table, ["scan", "get_item", "put_item", "delete_item"], context=context)
+        self._table = DynamoDBUtils.get_dynamodb_table_resource_ref(self._table_name)
```
Comment (CrypticCabub) [inline source/lambda/configuration/config_admin.py:111]: I don't see any reference to setting retries in this file, is standard retry the default now for the dynamo interface? (legacy is the default for clients - [doc](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html))

File: source/lambda/main.py [hunk 8]
```diff
@@ -30,21 +30,6 @@
...
-    if aws_data_path is not None:
-        aws_data_path = ":".join([aws_data_path, models])
-    else:
-        aws_data_path = models
-    os.environ["AWS_DATA_PATH"] = aws_data_path
-
-
-load_models()
-
-
```
Comment (CrypticCabub) [inline source/lambda/main.py:47]: I presume this has been confirmed as an unused method? It is not clear to me how it pertains to the old retry logic

File: source/lambda/schedulers/ec2_service.py [hunk 9]
```diff
@@ -74,8 +75,7 @@ class Ec2Service:
     EC2_STOPPING_STATES = {EC2_STATE_SHUTTING_DOWN, EC2_STATE_STOPPING, EC2_STATE_STOPPED}
     EC2_STARTING_STATES = {EC2_STATE_PENDING, EC2_STATE_RUNNING}
 
-    dynamodb = boto3.resource('dynamodb')
-    maintenance_table = dynamodb.Table(os.environ['MAINTENANCE_WINDOW_TABLE'])
+    maintenance_table = DynamoDBUtils.get_dynamodb_table_resource_ref(os.environ['MAINTENANCE_WINDOW_TABLE'])
```
Comment (CrypticCabub) [inline source/lambda/schedulers/ec2_service.py:78]: we should extract these env variables into a centralized enum registry of some kind, i dislike the magic strings that don't get tested until runtime -- adding as backlog item

**Review by CrypticCabub — COMMENTED** [review]
I don't see any tests that confirm that the retry logic is actually being correctly applied. This may be an in-built feature of the boto library now but I wonder if it's possible to write a test (maybe with moto?) that would confirm that the client_with_retries is actually attempting retries as expected

**Comment by CrypticCabub** [comment]
> @CrypticCabub testing default boto3 functionality is out of scope for this user story and solution, we rely on boto3 to perform as the their API reference documentation so that we rely on not just retries, but also handling security, HTTP request response.

Implementation details of boto3 are indeed out of scope, but whether or not retries are attempted at all (i.e. is the configuration being correctly picked up) is still something I would like a unit test for

**Comment by CrypticCabub** [comment]
let's include it as a backlog item for the final 1.5 release since we're working on fixing the testing anyway

**Review by CrypticCabub — APPROVED** [review]
